src/Solvers/WordSearchGrid.tsx

```diff
@@ -0,0 +1,180 @@
+  private _wordListInputText: string = '';
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      gridInputText: '',
+      wordListInputText: '',
+      output: [],
+    };
+  }
+
+  public componentDidMount() {
+    super.componentDidMount();
+
+    if (this._gridInputElement) {
+      this._gridInputElement.focus();
+    }
+  }
+
+  public render() {
+    return (
+      <div className="WordSearchGrid">
+        <Grid>
+          <Row>
+            <Col md={4}>
+              <FormControl
+                componentClass="textarea"
+                className="WordSearchList-input"
+                onChange={(event: FormEvent<FormControl>) => this.onListTextChanged(event)}
+                placeholder="Word List To Find"
+                value={this.state.wordListInputText}
+              />
+            </Col>
+
+            <Col md={8}>
+                <FormControl
+                  componentClass="textarea"
+                  className="WordSearchGrid-input"
+                  inputRef={(input: HTMLInputElement) => { this._gridInputElement = input; }}
+                  onChange={(event: FormEvent<FormControl>) => this.onGridTextChanged(event)}
+                  placeholder="Grid Text"
+                  value={this.state.gridInputText}
+                />
+            </Col>
+          </Row>
+
+          <pre className="WordSearchGrid-output">
+            {this.state.output}
+          </pre>
+        </Grid>
+      </div>
+    );
+  }
+
+  protected getLocalStorageKey() {
+    return 'WordSearchGrid';
+  }
+
+  protected onSaveState() {
+    return {
+      gridInputText: this._gridInputText,
+      wordListInputText: this._wordListInputText
+    };
+  }
+
+  protected onRestoreState(savedState: SavedState | null) {
+    if (savedState !== null) {
+      this._gridInputText = savedState.gridInputText;
+      this._wordListInputText = savedState.wordListInputText;
+    }
+  }
+
+  protected onUpdateState() {
+    this.setState({
+      gridInputText: this._gridInputText,
+      wordListInputText: this._wordListInputText,
+      output: this.calculateOutput(),
+    });
+  }
+
+  private onGridTextChanged(event: SyntheticEvent<FormControl>) {
+    const element = (event.target as HTMLInputElement);
+    this._gridInputText = element.value;
+    this.updateState();
+  }
+
+  private onListTextChanged(event: SyntheticEvent<FormControl>) {
+    const element = (event.target as HTMLInputElement);
+    this._wordListInputText = element.value;
+    this.updateState();
+  }
+
+  private onClearClick(event: MouseEvent<Button>) {
+    this._gridInputText = '';
+    this.updateState();
+  }
+
+  private toggleHomogeneous(event: MouseEvent<Button>) {
+    this.updateState();
+  }
+
+  private calculateOutput() {
+    if (!this._gridInputText) {
+      return [];
+    }
+    // get inputs
+    const lines = this._gridInputText.split(/\r?\n/);
+    const wordsToFind = this._wordListInputText.split(/\r?\n/);

```

Comment:
Probably want to trim whitespace off each word to find here.

Reply:
Actually, you may want to do a full up filtering of characters not found in the grid (or at least whitespace) to make it easier to search for phrases in a word search.